### PR TITLE
TSA lesson-based daml app infrastructure

### PR DESCRIPTION
**Pull request by cathyjung-da**
TSA lesson-based doc on daml app infrastructure is ready to be merged.

### CODE REVIEW

File: docs/3.1/docs/tsa-arch/daml-app-infrastructure.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
+Daml Application Infrastructure
+###############################
+
+Daml Application Infrastructure: Minimum Requirements to Run a Daml Application on Canton
+*****************************************************************************************
+Daml applications run on the Canton infrastructure and provide the benefits of distributed ledger technology, data sovereignty, and privacy. This documentation explores the components of Canton and the minimum requirements to run a Daml application on the Canton infrastructure.
+
+Infrastructure at a High-Level
+******************************
```
Comment (carrielaben-da) [inline docs/3.1/docs/tsa-arch/daml-app-infrastructure.rst:9]:
```suggestion
High-Level Infrastructure
*************************
```

File: docs/3.1/docs/tsa-arch/daml-app-infrastructure.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+Communication Between Participant Nodes and Synchronizers
+=========================================================
+Participant nodes communicate with synchronizers via the Canton protocol, which is implemented using gRPC.
+
+Participant nodes send two types of messages to the sequencer using the gRPC protocol:
+
+* Send: a regular message; a single gRPC message, to which the sequencer responds with a one-time response.
+* Read: a subscription request; a server-side streaming gRPC, to which the sequencer responds with a stream of messages.
+
+Both send and read operations are always initiated by participant nodes to the sequencer. Therefore, to enable required communication between infrastructure components, the sequencer must permit inbound gRPC connectivity, while the participant nodes must permit outbound gRPC connectivity.
```
Comment (carrielaben-da) [inline docs/3.1/docs/tsa-arch/daml-app-infrastructure.rst:41]:
```suggestion
Participant nodes always initiate both send and read operations to the sequencer. Therefore, the sequencer must permit inbound gRPC connectivity, while the participant nodes must permit outbound gRPC connectivity.
```

File: docs/3.1/docs/tsa-arch/daml-app-infrastructure.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+The ledger records cryptographic evidence of the workflow steps taken by both the app provider and the app users. Therefore, none of them can effectively repudiate or deny their steps in shared workflows. For example:
+
+* An app user cannot deny to the app provider that they submitted a buy order.
+* An app provider cannot deny to the user that they wrongfully canceled an order submitted by an app user.
+
+Non-repudiation applies to both the app provider and the app users.
+
+Self-sovereignty Over App Data for App Users
+============================================
+App users receive a copy of all their app data that is synchronized in real-time with the provider and other stakeholders’ copy of that data. This copy empowers app users to build custom integrations of their application data with their internal systems at will. Additionally, this copy implies that the data remains accessible even if the provider loses it, for example, as a result of a ransomware attack.
```
Comment (carrielaben-da) [inline docs/3.1/docs/tsa-arch/daml-app-infrastructure.rst:69]:
```suggestion
App users receive a copy of all their app data that is synchronized in real time with the provider and other stakeholders’ copies of that data. This copy empowers app users to build custom integrations of their application data with their internal systems at will. Additionally, this copy implies that the data remains accessible even if the provider loses it, for example, as a result of a ransomware attack.
```

File: docs/3.1/docs/tsa-arch/daml-app-infrastructure.rst [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+The app user’s participant node validates all changes to the app user’s data before they are committed. The validation checks are specified in the application’s DAR files uploaded to the user’s participant node. For example, if the application’s DAR file specifies that the app user must authorize all asset transfers, the user’s participant node ensures that this rule is enforced. This prevents third parties, including the app provider, from tampering with the record.
+
+Architecture and Control
+========================
+While it is technically possible to have an architecture where the app provider has access to an app user’s participant node, such architecture would necessarily result in the loss of the non-repudiation property and is therefore considered an anti-pattern. Each organization should use its own Identity and Access Management (IAM) system to control access to its participant node. This approach allows organizations to maintain control over their data, ensuring the system achieves the non-repudiation property.
```
Comment (carrielaben-da) [inline docs/3.1/docs/tsa-arch/daml-app-infrastructure.rst:75]:
```suggestion
While it is technically possible to have an architecture where the app provider has access to an app user’s participant node, such architecture would necessarily result in the loss of the non-repudiation property. Each organization should use its own Identity and Access Management (IAM) system to control access to its participant node. This approach allows organizations to maintain control over their data, ensuring the system achieves the non-repudiation property.
```

**Review by carrielaben-da — APPROVED** [review]
Small changes.

File: docs/3.1/docs/tsa-arch/daml-app-infrastructure.rst [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+Communication Between Participant Nodes and Synchronizers
+=========================================================
+Participant nodes communicate with synchronizers via the Canton protocol, which is implemented using gRPC.
+
+Participant nodes send two types of messages to the sequencer using the gRPC protocol:
+
+* Send: a regular message; a single gRPC message, to which the sequencer responds with a one-time response.
+* Read: a subscription request; a server-side streaming gRPC, to which the sequencer responds with a stream of messages.
+
+Both send and read operations are always initiated by participant nodes to the sequencer. Therefore, to enable required communication between infrastructure components, the sequencer must permit inbound gRPC connectivity, while the participant nodes must permit outbound gRPC connectivity.
```
Comment (cathyjung-da) [inline docs/3.1/docs/tsa-arch/daml-app-infrastructure.rst:41]:
```suggestion
Participant nodes always initiate both send and read operations to the sequencer. Therefore, the sequencer must permit inbound gRPC connectivity, while the participant nodes must permit outbound gRPC connectivity to enable required communication between infrastructure components.
```

**Pull request merged**